dotcom-rendering/src/components/YoutubeAtom/YoutubeAtomPlayer.tsx

```diff
@@ -83,12 +83,41 @@ const fullscreenStyles = (id: string) => css`
 		position: fixed;
 		top: 0;
 		left: 0;
-		width: 100vw;
-		height: 100vh;
-		z-index: ${getZIndex('youTubeFullscreen')};
+		width: 100svw;
+		height: 100svh;
+		${getZIndex('youTubeFullscreen')};
 	}
 `;
 
+/**
+ * We set the external_fullscreen configuration property depending on
+ * whether the native layer needs to delegate fullscreen styling to
+ * the webview.
+ *
+ * This is true for Android but not for iOS which handles fullscreen
+ * natively. The Bridget method setFullscreen returns a value to
+ * indicate this requirement. We use it here by passing a value of
+ * false for fullscreen when intialising the player to determine
+ * the value we need to pass for external_fullscreen.
+ *
+ * external_fullscreen is allowed listed on our CODE and PROD domains.
+ */
+const setAppsConfiguration = async (
+	basePlayerConfiguration: YouTubePlayerArgs,
+	renderingTarget: RenderingTarget,
+) => {
+	if (renderingTarget === 'Apps') {
+		const videoClient = getVideoClient();
+		const requiresWebFullscreen = await videoClient.setFullscreen(false);
+		const updatedConfiguration = {
+			...basePlayerConfiguration,
+			external_fullscreen: requiresWebFullscreen ? 1 : 0,
+		};
+		return updatedConfiguration;
+	}
+	return Promise.resolve(basePlayerConfiguration);
```

Comment:
Since this function is `async` can this be returned directly? Does it need to be wrapped in a `Promise.resolve`?

Reply:
Good spot thanks. Have updated